src/connector/src/source/reader/reader.rs

```diff
@@ -49,7 +48,7 @@ pub struct SourceReader {
 
 impl SourceReader {
     pub fn new(
-        properties: BTreeMap<String, String>,
+        properties: WithOptionsSecResolved,
```

Comment:
Continue from the https://github.com/risingwavelabs/risingwave/pull/17456#discussion_r1663820285, I want to limit the invasion of secret as much as possible. For example here, the `SourceReader` is possible to be entirely decoupled from secrets, so I'd like to `fill_secrets` outside this function.

Reply:
The reason I prefer using `WithOptionsSecResolved ` here is to avoid potential misuse of `BTreeMap<String, String>` and make sure no one ever try to give a `SourceReader::new` without secret.  (`BTreeMap<String, String>` should be expelled from our code base)

Reply:
> make sure no one ever try to give a `SourceReader::new` without secret.

Yeah, I am trying to avoid this too but in a slightly different approach: I want to eliminate anything related to secrets (incl. `WithOptionsSecResolved`) as soon as possible.

Let me elaborate my idea more. 

As discussed in https://github.com/risingwavelabs/risingwave/pull/17456#discussion_r1665142268, both ① protobuf message and ② the structs that supports `to_protobuf` must keep the `secret_ref` instead of plaintext.

**Except these 2 places, I'd like to keep everything unaware of secrets.** 

For example, here the calling path is 

```
stream::from_proto::source::create_source_desc_builder()
--> source::reader::SourceDescBuilder::new()
    --> SourceReader::new
```

I would propose to do `fill_secrets` in `create_source_desc_builder()` or earlier. As you can see, this function belongs to the `from_proto`, which indicates it's a conversion from ① (protobuf message) to something that is neither ① nor ②.

Reply:
Oh I see your point now. There are still 2 concerns
1. There are no concrete type for BTreeMap<String, String> with filled secret know. Leaving more `BTreeMap<String, String>` is a problem. We can definitely do this better after I add a type for that later.
2. For the `SourceDescBuilder` case. Leaving `WithOptionsSecResolved` without filling in secret give us ability to alter a secret for a stream job online without rebuilding the stream graph.